src/main/java/hudson/plugins/promoted_builds/parameters/PromotedBuildParameterDefinition.java

```diff
@@ -42,22 +44,42 @@
 import java.util.ArrayList;
 import java.util.Iterator;
 import java.util.List;
+import java.util.StringTokenizer;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
 import jenkins.model.Jenkins;
 import net.sf.json.JSONObject;
 import org.apache.commons.lang.StringUtils;
 import org.kohsuke.stapler.AncestorInPath;
 import org.kohsuke.stapler.DataBoundConstructor;
 import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.Stapler;
 import org.kohsuke.stapler.StaplerRequest;
 import org.kohsuke.stapler.export.Exported;
 
 /**
  * Defines a parameter that allows the user to select a promoted build
  * from a drop down list.
- *
+ * <p>
+ * Remarks on addressing: Starting from TODO, the field also supports folders (JENKINS-25011)
+ * We are using the absolute path from the root by default in order to retain the original behavior 
+ * of the plugin. Effectively, the resolution order is following:
+ * <ul>
+ *  <li>If an item for the name exists globally, it will be returned</li>
+ *  <li>If the path starts with &quot;/&quot;, a global addressing will be used</li>
+ *  <li>If the path starts with &quot;./&quot;, a relative addressing will be used</li>
+ *  <li>If there is no prefix, a relative addressing will be tried. If it fails, the method falls back to a global one</li>
```

Comment:
I think I see in the `getItemByPath` method that you only check the global if the path is prefixed with a slash, so it makes sense now i.e. there isn't really an order of precedence here, so ignore my previous comment :smile: